src/search.coffee

```diff
@@ -6,234 +6,252 @@ url = require 'url'
-    reqOptions =
-      Query: @quoted options.query
-      $top: options.top or 10
-      $skip: options.skip or 0
-
-    # Filter out unsupported sources
-    sources = (s for s in options.sources or [] when s in Search.SOURCES)
-    reqOptions.Sources = @quoted sources if sources.length
-
-    reqOptions.Market = @quoted(options.market) if options.market in markets
-    reqOptions
-
-  # Given a list of strings, generates a string wrapped in single quotes with
-  # the list entries separated by a `+`.
-  quoted: (values) ->
-    values = [values] unless _.isArray values
-    values = (v.replace "'", "''" for v in values)
-    "'#{values.join '+'}'"
-
-  # Generates a sequence of numbers no larger than the page size which the sum
-  # of the list equal to numResults.
-  generateTops: (numResults, pageSize = Search.PAGE_SIZE) ->
-    tops = [numResults % pageSize] if numResults % pageSize isnt 0
-    tops or= []
-    (pageSize for i in [0...Math.floor(numResults / pageSize)]).concat tops
-
-  # Generate a sequence of offsets as a multiple of page size starting at
-  # skipStart and ending before skipStart + numResults.
-  generateSkips: (numResults, skipStart) ->
-    skips = [skipStart]
-    for count in @generateTops(numResults)[...-1]
-      skips.push skips[skips.length - 1] + count
-    skips
-
-  parallelSearch: (vertical, options, callback) ->
-    opts = _.extend {}, {top: Search.PAGE_SIZE, skip: 0}, options
-
-    # Generate search options for each of the search requests.
-    pairs = _.zip @generateTops(opts.top), @generateSkips(opts.top, opts.skip)
-    requestOptions = _.map pairs, ([top, skip]) ->
-      _.defaults {top, skip}, options
-
-    search = (options, callback) =>
-      @search vertical, options, callback
-
-    async.mapLimit requestOptions, @parallel, search, callback
-
-  search: (vertical, options, callback) ->
-    requestOptions =
-      uri: "#{BING_SEARCH_ENDPOINT}/#{vertical}"
-      qs: _.extend @requestOptions(options),
-        $format: 'json'
-      auth:
-        user: @accountKey
-        pass: @accountKey
+  ###
+  The PAGE_SIZE variable is used for `count` on most API requests. The
+  documentation states that the maximum results is API specific; but, all of
+  the API references just state that "the actual number delivered may be less
+  than requested." From what I've seen, searches typically return ~30  results;
+  however, impartial result sets occur at any `count > 10`. `25` is a good
+  number for avoiding holes in data and duplicate results across pages.
+  ###
+  @PAGE_SIZE = 25
+
+  ###
+  The MAX_RESULTS variable, used for the legacy `top` pagination option, allows
+  for us to return the same default `50` results as v1.0.1 of this library.
+  ###
+  @MAX_RESULTS = 50
+
+  constructor: (@accountKey, @parallelLimit = 10, @useGzip = true) ->
+
+  sanitizeOptions: (options) ->
+    # Default pagination.
+    options = _.defaults options,
+      count: Search.PAGE_SIZE
```

Comment:
I thought so, was again copying Dudley's source. Will update because `@PAGE_SIZE` makes more sense to me.